### PR TITLE
Unregistered contributors now show up as contributors to a project in search

### LINKED ISSUE #1724: Unconfirmed contributors do not show up in search results for a project
<h4>Steps</h4>
1. Search the OSF for a project that has an unconfirmed contributor (ie. a non-OSF user added to it)
For example, search for "Halevy"

<h4>Actual Results</h4>
1. Only confirmed contributors show up in search results. For instance, if you visit the project, you see three contributors—one unconfirmed. The search results only show the two confirmed contributors.
screen shot 2015-02-05 at 10 13 26 am

screen shot 2015-02-05 at 10 13 30 am

<h4>Expected Results</h4>
1. I would expect that all contributors would show up in search results of projects, not just the confirmed contributors.

### PR DESCRIPTION
**Pull request by fabianvf**
## Purpose
Unregistered contributors who have been added to projects don't show up as contributors on those projects in the search results. When users find projects in the search results, the list of contributors doesn't reflect what the list of contributors is in the project overview page. This is potentially confusing.

## Changes
Now, unregistered users are listed as contributors to projects in the search results. These users still do not show up in the user search though.

## Side effects
None expected.

closes #1724

### CODE REVIEW

File: tests/test_elastic.py [hunk 1]
```diff
@@ -397,6 +397,18 @@ def test_unreg_users_dont_show_in_search(self):
         contribs = search.search_contributor(unreg.fullname)
         assert_equal(len(contribs['users']), 0)
 
+
+    def test_unreg_users_do_show_on_projects(self):
```
Comment (brianjgeiger) [inline tests/test_elastic.py]: While we're thinking of it, if we don't have the test, could you add one really quick to verify that unregistered contributors don't show up in user search? Obviously this code couldn't possibly do that, but who knows what people may do in the future.

Comment (fabianvf) [inline tests/test_elastic.py]: Yeah, I think that test is already in there (right above the one I added, though you can't see it in the diff)

**Comment by fabianvf** [comment]
@sloria

**Comment by sloria** [comment]
Thanks. :octopus:

**Pull request merged**